api/classes/base_controller.js

```diff
@@ -41,37 +41,25 @@ BaseController.prototype.create = function(req, reply) {
 };
 
 BaseController.prototype.update = function(req, reply) {
-  var current_state = this.Model.query({
+  var reqData = this.data(req);
+  var result = this.Model.query({
     where: {
       id: req.params.id
     }
-  }).fetch().then(function(record){
+  })
+  .fetch()
+  .then(function(record) {
     if (!record) { throw Boom.notFound(); }
-    return record.toJSON();
-  });
-  this.Model.query({
-    where: {
-      id: req.params.id
+    record.set(reqData);
+    if (!record.hasChanged()) {
+      throw Boom.create(404);
     }
+    return record
+      .save(record.changed, { patch: true, method: 'update' });
   })
-  .save(this.data(req), {
-    method: 'update',
-    patch: 'true'
-  });
-  var updated_state = this.Model.query({
```

Comment:
I have no idea how this worked before. All this stuff happening in parallel must have been a nightmare to debug.